lib/abstract_recommender.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+
+
+class AbstractRecommender(object):
+    """
+    A class that acts like an interface,
+    it is never initialized but the uv_decomposition
+    and content_based should implement it's methods
+    """
+
+    def __init__(self):
+        raise NotImplementedError("Can't initialize this class")
+
+    def train(self):
+        raise NotImplementedError("Can't call this method")
+
+    def split(self):
```

Comment:
Do we need this in the abstract class? as we discussed shortly, not all recommenders will need a split.